### ISSUE TITLE
Relicensing to LGPL

### ISSUE THREAD

**Issue by marado** [tbamud/tbamud#95]
Following the re-licensing of DikuMUD into LGPL, CircleMUD has also changed its license to LGPL, opening the doors for tbamud to do the same, and finally have a free software license.

https://www.circlemud.org/license.html

Will you please consider doing so?

**Comment by krinosx** [comment]
Please dont...

LGPL is a terrible license to a MUD Codebase. If you guys want to license,
use GPL, Apache or other permissive license.

LGP is a good license if you wish to license a library or anything like
that. As you know, when you develop muds using LGP you are forced to also
opensource your own codebase even if you are doing minor/cosmetic changes.

The problem is not about releasing your changes, but you must host the code
and ensure you have a link and a disclaimer in your own site/mud always
pointing to it.

The guys on Circle Mud argue that now you can even release commercial mud
using LGPL, but it's not the case if you have to release and make all your
code open.


Anyway, think better if you guys are trying to use a OpenSource license ,
LGPL is not the better one for a MUD.

**Comment by belisariussmith** [comment]
While I agree with you, unfortunately, because tbaMUD is based on CircleMUD, it can only choose to limit itself with the older more restricted license, or what CircleMUD has moved to, the LGPL. Circle is likewise restricted by the Diku-base, which recently upgraded from its draconian license to the LGPL (which is the only reason why Circle was able to 'unlimit' itself in turn).Do I wish Diku has chosen to go with the MIT license? Of course I do.. however, even moving to LGPL is a massive step forward. Hopefully one day they will see reason and move to a better license.Yours Truly,BelisariusBelisarius A Smith | Consultant | BSBA, MSE, MPsy | www.belisariussmith.com 
-------- Original message --------From: Giuliano Bortolassi <notifications@github.com> Date: 8/23/20  9:01 PM  (GMT-06:00) To: tbamud/tbamud <tbamud@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: Re: [tbamud/tbamud] Relicensing to LGPL (#95) 

Please dont...



LGPL is a terrible license to a MUD Codebase. If you guys want to license,

use GPL, Apache or other permissive license.



LGP is a good license if you wish to license a library or anything like

that. As you know, when you develop muds using LGP you are forced to also

opensource your own codebase even if you are doing minor/cosmetic changes.



The problem is not about releasing your changes, but you must host the code

and ensure you have a link and a disclaimer in your own site/mud always

pointing to it.



The guys on Circle Mud argue that now you can even release commercial mud

using LGPL, but it's not the case if you have to release and make all your

code open.





Anyway, think better if you guys are trying to use a OpenSource license ,

LGPL is not the better one for a MUD.









On Sun, Aug 23, 2020 at 10:20 PM Marcos Marado <notifications@github.com>

wrote:

**Comment by fungi** [comment]
LGPL requires you to offer to provide source code for your modified MUD for access by anyone to whom you also distributed a compiled copy, and would allow anyone who received a copy of the source code to modify and redistribute it to others under the same terms.

If your desire is to make money by selling compiled copies of your MUD to customers and then suing any who redistributed it for copyright ifringement, then yes LGPL is not a fit. If on the other hand you want to run a MUD and charge for accounts on it, LGPL allows that without requiring you to publish your source code.

The LGPL does **not** force you to "open source" (publish) your codebase unless you are **also** distributing compiled copies of the MUD, and even then you only have to provide your source code to people to whom you provided the compiled version.

**Comment by moshehbenavraham** [comment]
amazing!  our version is public and we don't charge for anything, so i think we're covered...  but this is an amazing change to the license




Apex Web Services <https://apexwebservices.com/>
<https://moshehwebservices.com/>
Mesilat Yosef 1
Modiin Ilit, Israel 7182028

We keep office hours Sunday-Thursday
8:00am – 2:00pm Eastern Time Zone
Israel: (052) 593 6481<tel:0525936481>
USA: (619) 780-4648

________________________________

**Comment by fungi** [comment]
Well, per discussion on the CircleMUD ML and the TbaMud forum, TbaMud is not yet able to be distributed under the LGPL (and may never be). While it's true that DIKUMUD and CircleMUD are officially available under the LGPL since last year, TbaMud incorporates a number of fairly old patches on top of CircleMUD (things like ASCII PFiles, DG Scripts, Oasis OLC) whose authors would need to be reached for permission to relicense their work.

Tracking down these individuals and asking them would be the next step toward relicensing.

**Comment by moshehbenavraham** [comment]
Okay thanks Jeremy, good to know.  Is this something we should actively be pursuing?  I don't foresee a need for our project currently, but I do like the idea of people cloning our project and having less restrictions.




Apex Web Services <https://apexwebservices.com/>
<https://moshehwebservices.com/>
Mesilat Yosef 1
Modiin Ilit, Israel 7182028

We keep office hours Sunday-Thursday
8:00am – 2:00pm Eastern Time Zone
Israel: (052) 593 6481<tel:0525936481>
USA: (619) 780-4648

________________________________

**Comment by fungi** [comment]
Well, I can't tell you what you should or shouldn't be pursuing, I was mostly trying to clear up some misconceptions about open source and copyleft licenses (and particularly the LGPL) in earlier responses from others on this issue.

My perspective, as someone who's long wanted to see a DIKU descendant included in the Debian GNU/Linux distribution, TbaMud is the best maintained one with the most active remaining community. The old DIKU license basically prevented officially adding any DIKU-derived MUDs to Debian, but with it relicensed to the LGPL that would now be possible. Unfortunately, the DIKUMUD and even CircleMUD codebases require a lot of patching to get them compiling with a modern cc and libc, and they're not as feature-rich as TbaMud owing to almost 20 years of further development and refining.

If TbaMud were able to relicense to LGPL, I'd probably work on packaging it for inclusion in Debian. As it stands now, that's not possible, and I'm not sure I personally have the time to track down old patch authors. As mentioned above, some related discussion (unfortunately both require subscriber logins):

https://lists.queensu.ca/cgi-bin/listserv/wa?A1=ind2007&L=circle
https://www.tbamud.com/forum/2-general/4604-tbamud-license-change

**Comment by welcor** [comment]
Indeed, we've started the work in tracking down the people who have contributed over time. Quite a few of the authors of the patches have replied back with their approval (so far we've seen little negative feedback once the meaning of the LGPL has been explained, as above). However, efforts have been mostly halted since about August 2020 due to other commitments eating into the maintainers' time.  
I've tried tracking down a couple of developers who we have little information about (a mail address and a mud name, basically) via the circlemud mailing list, unfortunately with slim results to show for it.

The tbamud.com link should now work for non-registered users. I believe access was blocked due to spam bots. We'll see if they come back now I've opened for reading again.

**Comment by tjr1974** [comment]
Perhaps the easiest solution is to use a statement (disclaimer) that other portions are copyright by authors as noted in ChangeLog and source code. That's how Jeremy Elson handled it in the original CircleMUD license:

   Copyright (C) 1995 by Jeremy Elson
   CircleMUD 3 -- Copyright (C) 1994-2001, The CircleMUD Group
   Other portions copyright by authors as noted in ChangeLog and source code.

This way tbaMUD owned/created code is licensed LGPL  and copyright for code by other authors remain covered by the original CircleMUD liscense or whatever terms the authors choose to exercise. As I understand it, this is the entire point of the LGPL.

**Comment by fungi** [comment]
The point is to get all the code distributed as part of TbaMUD available under an OSI-approved license (so that, for example, it can be included in GNU/Linux distributions).

TbaMUD incorporates patches like OasisOLC, DGScripts, ASCII PFiles, and so on which were previously distributed by their respective authors independent of the CircleMUD codebase, and would need to be contacted for permission. Even just relicensing original contributions for the TbaMUD project would need the agreement of each contributor who wrote any of it.

All of those bits were distributed under the old DIKU/Circle license rather than the LGPL, so the hard part (which welcor mentioned) is reaching everyone who ever wrote those things under the old license to get them to agree (or replacing what they provided with wholly new code, which is tricky to do since commits which change that code are also derivative of the previous work and therefore would need to be redone, and all without looking at the code being replaced so as to avoid being unduly influenced by it in ways which could amount to a copyright violation).

It's a very sticky legal situation which other projects with similarly ancient lineage rarely manage to solve, especially if some authors have died and you end up needing to seek permission from their heirs.

**Comment by tjr1974** [comment]
> The point is to get all the code distributed as part of TbaMUD available under an OSI-approved license (so that, for example, it can be included in GNU/Linux distributions).

Ah, yes, that's an enitrely different circumstance from simply licensing ttbaMUD owned/created code as LGPL . :(

**Comment by pdub206** [comment]
Is this still under consideration?

**Comment by welcor** [comment]
In short, yes. See https://www.tbamud.com/kunena/2-general/4604-tbamud-license-change?start=12#10955
